cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -277,6 +277,21 @@ func deviceLayoutFromPartitionTable(ptable sfdiskPartitionTable) (*DeviceLayout,
 			Type:       p.Type,
 			Filesystem: bd.FSType,
 		}
+
+		// sometimes sfdisk reports what is "0C" in gadget.yaml as "c", so fix
+		// that up now
+		// TODO:UC20: what's a better way?
+		if structure[i].Type == "c" {
+			structure[i].Type = "0C"
+		}
+
+		// sometimes sfdisk reports no "Name" for partitions, so in this case
```

Comment:
Unfortunately MBR partition entries have no name field so the label is all we have.